lib/languages/sass.js

```diff
@@ -0,0 +1,180 @@
+var util = require('util');
+var DocsParser = require("../docsparser");
+var xregexp = require('../xregexp').XRegExp;
+
+function SassParser(settings) {
+    // this.setup_settings();
+    // call parent constructor
+    DocsParser.call(this, settings);
+}
+
+SassParser.prototype = Object.create(DocsParser.prototype);
+
+SassParser.prototype.setup_settings = function() {
+    var identifier = '[a-zA-Z_-][a-zA-Z_0-9-]*';
+    this.settings = {
+        'commentType': 'single',
+        'curlyTypes': true,
+        'typeInfo': true,
+        'typeTag': 'type',
+        'prefix': '///',
+        'varIdentifier': '\\$' + identifier,
+        'fnIdentifier':  identifier,
+        'fnOpener': '@(?:mixin|function)\\s+' + identifier + '\\s*[\\(|\\{]',
+        'bool': 'Boolean'
+    };
+};
+
+SassParser.prototype.parse_function = function(line) {
+    var r = '^\\s*@(?P<fnType>(mixin|function))\\s+' +
+            '(?P<name>' + this.settings.fnIdentifier + ')' +
+            '\\s*(?:\\(\\s*(?P<args>.*?)\\)|{)';
+    var regex = xregexp(r);
+
+    var matches = xregexp.exec(line, regex);
+    if(matches === null)
+        return null;
+
+    return [
+        matches.name,
+        (matches.args ? matches.args.trim() : null),
+        (matches.fnType == "mixin" ? null : false)
+    ];
+};
+
+SassParser.prototype.parse_var = function(line) {
+    //   $foo: blah;
+    var r = '(?P<name>' + this.settings.varIdentifier + ')' +
+            '\\s*:\\s*(?P<val>.*?)(?:;|$)';
+    var regex = xregexp(r);
+
+    var matches = xregexp.exec(line, regex);
+
+    if(matches === null) {
+        return null;
+    }
+    // variable name, variable value
+    return [matches.name, matches.val.trim()];
+};
+
+SassParser.prototype.parse_arg = function(arg) {
+
+    var regex = xregexp(
+        '(?P<name>' + this.settings.varIdentifier + ')(\\s*:\\s*(?P<value>.*))?'
+    );
+
+    return xregexp.exec(arg, regex);
+};
+
+SassParser.prototype.get_arg_type = function(arg) {
+
+    var matches = this.parse_arg(arg);
+
+    if(matches && matches.value) {
+        return this.guess_type_from_value(matches.value);
+    }
+
+    return null;
+};
+
+SassParser.prototype.get_arg_name = function(arg) {
+
+    var matches = this.parse_arg(arg);
+
+    if (matches && matches.value) {
+      return util.format('%s [%s]', matches.name, matches.value);
+    }
+
+    return matches.name;
```

Comment:
In the `if` statement above it ensures that matches is not `undefined`. Something like this should be done for the return statement too. Otherwise entering invalid parameters will result in a exception. E.g. `function foo(bar) {}` (no `$`). Ideally also add a test for this.